Mailer/app/api/v1/Handlers/Users/UserPreferencesHandler.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Remp\MailerModule\Api\v1\Handlers\Users;
+
+use Remp\MailerModule\Api\Validation\InputValidatorTrait;
+use Remp\MailerModule\Repository\UserSubscriptionsRepository;
+use Remp\MailerModule\Repository\UserSubscriptionVariantsRepository;
+use Tomaj\NetteApi\Handlers\BaseHandler;
+use Tomaj\NetteApi\Response\JsonApiResponse;
+
+class UserPreferencesHandler extends BaseHandler
+{
+    private $userSubscriptionsRepository;
+
+    private $userSubscriptionVariantsRepository;
+
+    use InputValidatorTrait;
+
+    public function __construct(
+        UserSubscriptionsRepository $userSubscriptionsRepository,
+        UserSubscriptionVariantsRepository $userSubscriptionVariantsRepository
+    ) {
+        parent::__construct();
+        $this->userSubscriptionsRepository = $userSubscriptionsRepository;
+        $this->userSubscriptionVariantsRepository = $userSubscriptionVariantsRepository;
+    }
+
+    public function handle($payload)
+    {
+
+        $request = file_get_contents("php://input");
+        $payload = $this->validateInput($request, 'user-preferences.schema.json');
+
+        if ($this->hasErrorResponse()){
+            return $this->getErrorResponse();
+        }
+
+//        $rows = $this->userSubscriptionsRepository->findByEmail($payload['email']); TODO use later
+
+        $rows = $this->userSubscriptionsRepository
+            ->getTable()
+            ->where(['user_id' => $payload['user_id'], 'user_email' => $payload['email']]);
+
+
+        if (isset($payload['subscribed'])){
+            $rows->where('subscribed', $payload['subscribed']);
+        }
+
+        $rows = $rows->fetchAll();
+
+        $output = [];
+
+        $subscribedVariants = [];
+
+        foreach ($this->userSubscriptionVariantsRepository->multiSubscribedVariants($rows) as $variant) {
+            $subscribedVariants[$variant->mail_user_subscription_id][] = $variant;
+        }
+
+        foreach ($rows as $row) {
+            $variants = [];
+            foreach ($subscribedVariants[$row->id] ?? [] as $variant) {
+
+                $variants[] = [
+                    'id' => $variant->mail_type_variant_id,
+                    'code' => $variant->code,
+                    'title' => $variant->title,
+                ];
+            }
+
+            $mailType =  $row->ref('mail_types', 'mail_type_id');
```

Comment:
You can just use `$row->mail_type` in the `$output` array below, no need to define this variable. It works the same way.